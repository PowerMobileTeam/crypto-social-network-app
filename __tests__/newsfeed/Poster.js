import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';

import imagePicker from '../../src/common/services/image-picker.service';
import Poster from '../../src/newsfeed/Poster';
import UserStore from '../../src/auth/UserStore';

import { 
  FlatList,
  KeyboardAvoidingView 
} from 'react-native';

import renderer from 'react-test-renderer';

jest.mock('../../src/auth/UserStore');

import * as dependency from '../../src/newsfeed/NewsfeedService';

fdescribe('Newsfeed poster component', () => {

  let user, entity, screen, post;
  beforeEach(() => {
    user = new UserStore();
    screen = shallow(
      <Poster.wrappedComponent user={user} guid='guid1' isRemind={false}/>
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    await screen.instance().componentWillMount();
    screen.update();
    expect(screen).toMatchSnapshot();
  });

  it('should have a TextInput', async () => {
    await screen.instance().componentWillMount();
    screen.update();
    let render = screen.dive();

    expect(render.find('TextInput')).toHaveLength(1);
  });

  it('should have a poster button', async () => {
    await screen.instance().componentWillMount();
    screen.update();
    
    let render = screen.dive();
    expect(render.find('TouchableHighlight')).toHaveLength(1);
  });

  it('should calls post on posting', async () => {

    const spy = jest.spyOn(dependency, 'post');

    const render = screen.dive();
    render.find('TextInput').forEach(child => {
      child.simulate('changeText', 'data');
    });

    await render.find('TouchableHighlight').at(0).simulate('press');

    expect(screen.instance().state.text).toEqual('data');

    expect(spy).toBeCalled();
  });

  it('should calls remind on posting a remind', async () => {

    user = new UserStore();
    screen = shallow(
      <Poster.wrappedComponent user={user} guid='guid1' isRemind={true}/>
    );

    jest.runAllTimers();

    const spy = jest.spyOn(dependency, 'remind');

    const render = screen.dive();
    render.find('TextInput').forEach(child => {
      child.simulate('changeText', 'data');
    });

    await render.find('Icon').at(0).simulate('press');

    expect(screen.instance().state.text).toEqual('data');

    expect(spy).toBeCalled();
  });

});
