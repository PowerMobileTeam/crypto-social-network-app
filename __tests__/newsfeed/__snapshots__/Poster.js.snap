// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsfeed poster component renders correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 7,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Speak your mind..."
        placeholderTextColor="#ccc"
        selectTextOnFocus={true}
        style={
          Object {
            "flex": 5,
            "minHeight": 40,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </Component>
    <Component
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Icon
        allowFontScaling={false}
        name="md-camera"
        onPress={[Function]}
        size={24}
      />
    </Component>
  </Component>
  <Component
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 5,
        }
      }
    />
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TouchableHighlight
        accessibilityLabel="Submit post"
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "#4690D6",
            "borderRadius": 5,
            "borderWidth": 1,
            "margin": 4,
            "padding": 5,
          }
        }
        underlayColor="transparent"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#4690D6",
            }
          }
        >
           POST 
        </Text>
      </TouchableHighlight>
    </Component>
  </Component>
  <Component
    style={
      Object {
        "flex": 5,
        "flexDirection": "row",
      }
    }
  >
    <Component />
    <Component />
  </Component>
</Component>
`;
